### PR TITLE
[DNM] Mecha rewrite.

### PR DESCRIPTION
**Pull request by Zuhayr**
leg day
new HUD
new mechs
decals

WIP
- [x] Add mech HUD.
- [x] Add mech construction.
- [x] Add modular mech icons/weapon overlays.
- [x] Reimplement decals.
- [ ] Work out how to get hardpoints to layer under the mech based on direction.
- [x] Fix bug with attacking windows on the edge adjacent to the mech.
- [x] Fix bug with flashlight.
- [x] Readd mech software/circuits.
- [x] Readd internal atmospherics.
- [x] Readd internal radio.
- [ ] Readd control consoles/beacons.
- [x] Add life/death.
- [x] Add wreckage and salvaging.
- [x] Add emp_act(), stun, sleep, paralysis handling.
- [x] Reimplement the means of producing mech parts.
- [x] Finalize and polish up the construction process.
- [ ] Finalize hacking minigame.
- [x] Add sounds to the construction deconstruction stuff.
- [ ] Test test test test test.

### CODE REVIEW

**Comment by mwerezak** [comment]
Looks pretty cool.

**Comment by Dorsisdwarf** [comment]
Jumping in from /tg/ to hail you as a god.

**Comment by NobleCaos** [comment]
Looks good, Zuh

**Comment by Zuhayr** [comment]
Going to work on this after Uni is over in a week or two, hasn't been abandoned.

**Comment by SkowronX** [comment]
Gimme some time and I'll have 3 new mech types sprited for this.

**Comment by SkowronX** [comment]
By the way what is the layer hierarchy for the mech parts?

**Comment by NobleCaos** [comment]
@Zuhayr I'll see if I can't fix that mapping bug... if it isn't already

**Comment by Carbonhell** [comment]
Any news on this?i'm hella hyped,looks so cool

**Comment by Yonaguni** [comment]
Still going to finish it off. Closing the PR and reopening to test the webhooks pointing to the new server.

**Comment by Zuhayr** [comment]
Got this up to date and fixed the odd atmos issues, hopefully going to have something mergable within a few days (ha ha).

**Comment by Zuhayr** [comment]
Going to go ahead and merge these now (if Travis passes), it's still a WIP but so is the entire branch. Don't port them yet, upstream.

**Comment by Memendia** [comment]
holy fucking shit this is amazing :+1:

File: code/modules/mecha/components/_components.dm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+/obj/item/mech_component/emp_act(var/severity)
+	take_burn_damage(rand((10 - (severity*3)),15-(severity*4)))
+	for(var/obj/item/thing in contents)
+		thing.emp_act(severity)
+
+/obj/item/mech_component/New()
+	..()
+	set_dir(SOUTH)
+
+/obj/item/mech_component/set_dir()
```
Comment (seekin4u) [inline code/modules/mecha/components/_components.dm]: set_dir(direction)
  ..(direction) ??

**Pull request merged**